cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/replication/ReplicationService.java

```diff
@@ -153,18 +157,25 @@ public void onCreate() {
         // Get the HandlerThread's Looper and use it for our Handler.
         Looper serviceLooper = thread.getLooper();
         mServiceHandler = getHandler(serviceLooper);
+    }
 
-        Replicator[] replicators = getReplicators(getApplicationContext());
-
+    public void setReplicators(Replicator[] replicators) {
```

Comment:
What is the expected behaviour if this method is called twice? I'm a bit concerned that it appears to be a setter, but at present it will add additional replicators. I'm not sure what happened in the previous implementation if the `Replicator[]` returned from `getReplicators()` was modified.
I think we need to do one of these three things:

1. Don't allow the list to be changed and throw an `IllegalStateException` or similar if `setReplicators` is called twice.
2.  Have setter behaviour by creating a new `ReplicationPolicyManager` (or clear the existing replicators before calling `mReplicationPolicyManager.addReplicators` although it doesn't look like there is a way to do this at present).
3. Rename the method to `addReplicators` and allow replicators to be added.

I think for cases 2 or 3 some considerations may need to be made for how the command queue is processed if commands effectively apply to different groups of replicators.